haskell-style.md

```diff
@@ -660,3 +660,329 @@ module Driver
     ) where
 
   ```
+
+## Haddocks
+
+Haskell modules should be commented with valid Haddock documentation. We are not
+yet requiring a certain level of coverage, but it is strongly encouraged.
+
+### General guides
+
+- Use proper Haddock [markup][]
+- Link all identifiers, anywhere they appear
+
+  Bad
+
+  ```hs
+  -- | Construct a @'FlipFlop'@
+  --
+  -- If the size is right, you will get a @Right FlipFlop@, otherwise a @Left@
+  --
+  ```
+
+  Good
+
+  ```hs
+  -- | Construct a @'FlipFlop'@
+  --
+  -- If the size is right, you will get a @'Right' 'FlipFlop'@, otherwise a
+  -- @'Left'@.
+  --
+  ```
+
+- Use leading documentation (`-- |`) for top-level definitions and trailing
+  documentation (`-- ^`) for record attributes and function arguments
+
+  Bad
+
+  ```hs
+  data Foo = Foo
+    {
+    -- | Foo's foo
+      fooFoo :: Foo
+    -- | Foo's bar
+    , fooBar :: Bar
+    }
+  -- ^ What even is a Foo?
+  ```
+
+  Good
+
+  ```hs
+  -- | What even is a Foo?
```

Comment:
Indeed

Reply:
A convenient mispelling of `fu` to avoid detection when coupled with `bar`?